src/redux/actions/articleActions.js

```diff
@@ -45,6 +46,12 @@ export const getArticleBySlug = slug => async (dispatch) => {
   dispatch(articleLoading());
   try {
     const fetchArticle = await API_SERVICE.get(`/articles/${slug}`);
+    const { tagsList } = fetchArticle.data.article;
+    if (!isEmpty(tagsList)) {
+      const randomTag = tagsList[Math.floor(Math.random() * tagsList.length)];
+      const relatedArticles = await API_SERVICE.get(`/search?keyword=${randomTag}`);
+      fetchArticle.data.article.relatedArticles = relatedArticles.data.matches.tags;
+    }
     dispatch(getSingleArticle(fetchArticle.data.article));
```

Comment:
I know you didn't make this change now but these just caught my attention. Ss the action `getSingleArticle` really getting a single article or has the article already been fetched and you are now updating the store with the value?

Also, the variable name `fetchArticle`, is that the best name for the **result** of fetching an article by slug?

Reply:
The getSingleArticle Action updates the store with the fetched article, and I'll rename the `fetchArticle` variable to something a bit more descriptive.

Reply:
`getSingleArticle` sounds like that's the guy that makes the request to the server, I don't know if you get